File: GXpert/GXpert.Web/Modules/Users/Teacher/TeacherColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Users.Columns;

[ColumnsScript("Users.Teacher")]
[BasedOnRow(typeof(TeacherRow), CheckNames = true)]
public class TeacherColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Prn { get; set; }
    public string Name { get; set; }
    public string Email { get; set; }
    public string Mobile { get; set; }
    public string Username { get; set; }
    public string Description { get; set; }
    public string StateTitle { get; set; }
    public string DistrictTitle { get; set; }
    public string TalukaTitle { get; set; }
    public DateTime Dob { get; set; }
    public string SchoolName { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
    public bool IsActive { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Users/TeacherColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TeacherRow } from "./TeacherRow";

export interface TeacherColumns {
    Id: Column<TeacherRow>;
    Prn: Column<TeacherRow>;
    Name: Column<TeacherRow>;
    Email: Column<TeacherRow>;
    Mobile: Column<TeacherRow>;
    Username: Column<TeacherRow>;
    Description: Column<TeacherRow>;
    StateTitle: Column<TeacherRow>;
    DistrictTitle: Column<TeacherRow>;
    TalukaTitle: Column<TeacherRow>;
    Dob: Column<TeacherRow>;
    SchoolName: Column<TeacherRow>;
}

export class TeacherColumns extends ColumnsBase<TeacherRow> {
    static readonly columnsKey = 'Users.Teacher';
    static readonly Fields = fieldsProxy<TeacherColumns>();
}